greedy_max/__init__.py

```diff
@@ -43,20 +52,84 @@ def schedule_observation(self, site: Site, obs_idx: int, start: int, finish: int
         from start to finish.
         """
         self.schedule[site][start:finish] = obs_idx
+    
+    def schedule_observations(self, site: Site, science: List[Observation], start: int, end: int) -> NoReturn:
+        """
+        Add all the observations inside a unit in the time slots from start to finish.
+        """
+        for sci in science:
+            if start < end:
+                    sci_length = end - start+ 1  if start + sci.length - 1 > end else sci.length
+                    self.schedule_observation(site, sci.idx, start, start + sci_length)
+                    sci.observed+=sci_length 
+                    start+=sci_length
 
     def is_observation_scheduled(self, site: Site, obs_idx: int) -> bool:
         """
         Determine if an observation is partially or fully scheduled at the specified site.
         """
         return len(self.slots_for_observation(site, obs_idx)) != 0
 
+    def is_complete(self, site: Site) -> bool:
+        return len(self._empty_slots(site)) < 0
+
+    def _intervals(self, empty_slots: np.ndarray) -> np.ndarray:
+        """
+        Given the empty slots in the schedule, returns an array with the indices with the
+        intervals that can schedule an observation
+        """
+        ni = len(empty_slots)
+        cvec = np.zeros(ni, dtype=int)
```

Comment:
`ni` is obvious number of intervals. What is a `cvec`?

Reply:
I agree with both. This is legacy code from the original version, so it would be useful to clarify what the variables mean because I'm not to clear either (that is why I left it as it was).